src/main/java/duke/task/Task.java

```diff
@@ -0,0 +1,82 @@
+package duke.task;
+
+/**
+ * The Task class represents a general task with a description and a completion status.
+ * It is a base class for specific task types like Todo, Deadline, and Event.
+ */
+public class Task {
+    protected String description; // Changed to private for encapsulation
+    protected boolean isDone; // Changed to private for encapsulation
+
+    /**
+     * Constructs a Task object with a description and initializes it as not done.
+     *
+     * @param description A description of the task.
+     */
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    /**
+     * Marks the task as done by setting the completion status to true.
+     */
+    public void markAsDone() {
+        this.isDone = true;
+    }
+
+    /**
+     * Unmarks the task as done by setting the completion status to false.
+     */
+    public void unmarkAsDone() {
+        this.isDone = false;
+    }
+
+    /**
+     * Returns the status number of the task (0 for not done, 1 for done).
+     *
+     * @return A string representing the task's completion status.
+     */
+    public String getStatusNumber() {
```

Comment:
Method name implies that the return type is a number but actual return type is a string.

Reply:
Thanks for pointing out. I have now changed the String Type to Integer Type.